core/base/inc/TROOT.h

```diff
@@ -297,6 +299,7 @@ friend TROOT *ROOT::Internal::GetROOT2();
    void              SaveContext();
    void              SetApplication(TApplication *app) { fApplication = app; }
    void              SetBatch(Bool_t batch = kTRUE) { fBatch = batch; }
+   void              SetRandomAccessCompression(Bool_t random = kTRUE) { fRandomAccessCompression = random; }
```

Comment:
This setting concerns only TTree and would be better placed in TTree as a static.   In addition we should support the case where some TTree have this setting on and some do not (in the same process), so this setting should *also* be settable for each TTree (Actually it looks like you already did that).